types/stylelint/index.d.mts

```diff
@@ -13,6 +13,7 @@ export {
 	DisabledRange,
 	DisabledRangeObject,
 	DisabledWarning,
+	FixerData,
```

Comment:
Do we need to bikeshed the new API?

Reply:
Not really, this is internal and follows what we have done so far for similar data.
The real bikeshedding will be about the details of the exposition of `range` and `fixed` in JSON.

e.g.
do we create `text` from `range`?
what should be the shape of the object?
etc.

In short this is on topic for #7192 and #4664.

Reply:
Thinking about it. You have a point.
From a user standpoint, at this stage having `Range` is more useful than `FixerData`.
i.e. the only usage in JSDoc is `FixerData['range']`

change incoming

Reply:
Before making further changes we should design this :)
Maybe the current form is the best?

--------

A way to get around this, at this time, is to not expose the `FixerData` and to limit the API surface additions to the `fix()` function.

Can we add the `fix()` function, collect the `FixerData` but without exposing any other new API surface or types than `fix()`?

Maybe even preferable to not collect the `FixerData` just yet?

The main purpose is to make this change easier to review and to find consensus on all parts :)

Reply:
This PR is a precursor for 3 issues by design.
If we want to decide to start working on any of the 3, we can.
It's meant to be an unblocker.

i.e.

If you add a refactor, you have the expectation related to the `throw` that we added.
So if anyone fixes #7192 while we have partially migrated the core rules, it will be available.
Same thing for #4664.
`Range` won't change and it's useful so it's OK to expose.
BTW `FixerData` is not exported anymore.

I could maybe not expose it if JSDoc supports something crazy like `(typeof StylelintPostcssResult['FixersData'][keyof typeof StylelintPostcssResult['FixersData']])[number]['range']`.
IMHO it's not worth the effort.